hack/ginkgo-e2e.sh

```diff
@@ -166,20 +175,41 @@ fi
 PATH=$(dirname "${e2e_test}"):"${PATH}"
 export PATH
 
-# Choose the program to execute.
-program=("${ginkgo}")
-if [[ "${E2E_TEST_DEBUG_TOOL}" == "delve" ]]; then
-  program=("dlv" "exec")
-elif [[ "${E2E_TEST_DEBUG_TOOL}" == "gdb" ]]; then
-  program=("gdb")
+# Choose the program to execute and additional arguments for it.
+#
+# Note that the e2e.test binary must have been built with "make DBG=1"
+# when using debuggers, otherwise debug information gets stripped.
+case "${E2E_TEST_DEBUG_TOOL:-ginkgo}" in
+  ginkgo)
+    program=("${ginkgo}" "${ginkgo_cli_args[@]}")
+    if [[ -n "${GINKGO_PARALLEL_NODES:-}" ]]; then
+      program+=("--nodes=${GINKGO_PARALLEL_NODES}")
+    elif [[ ${GINKGO_PARALLEL} =~ ^[yY]$ ]]; then
+      program+=("--nodes=25")
+    fi
+    program+=("${ginkgo_args[@]:+${ginkgo_args[@]}}")
+    ;;
+  delve) program=("dlv" "exec") ;;
+  gdb) program=("gdb") ;;
+  *) kube::log::error_exit "Unsupported E2E_TEST_DEBUG_TOOL=${E2E_TEST_DEBUG_TOOL}" ;;
+esac
+
+# Move Ginkgo arguments that are understood by the suite when not using
+# the CLI.
+suite_args=()
+if [ "${E2E_TEST_DEBUG_TOOL}" != "ginkgo" ]; then
+  for arg in "${ginkgo_args[@]}"; do
+    suite_args+=("--ginkgo.${arg#--}")
```

Comment:
can we have compatibility issues with this? I see that we require bash so it should be fine, right?

Reply:
`${arg#--}` to strip the prefix of `arg` is even supported by POSIX. Should be fine.